react-js/src/App.tsx

```diff
@@ -436,7 +436,13 @@ export default class App extends React.Component<any, any> {
 
   formatBarcodes(codes: Barcode[]): string {
     return JSON.stringify(
-      codes.map((code: Barcode) => code.text + " (" + code.format + ") ")
+      codes.map((code: Barcode) => {
+        if (code.parsedText) {
+          return code.parsedText;
+        } else {
+          return code.text + " (" + code.format + ") ";
+        }
+      })
     );
```

Comment:
Shouldn't be a huge problem, but there's an extra space after `)` in the string format defined in the else branch of the if.

Also, if you wish you can simplify the whole thing:

```typescript
codes.map((code: Barcode) => code.parsedText || `${code.text} (${code.format})`);
```